### ISSUE TITLE
setting __proto__ in an object literal does not behave as expected

### ISSUE THREAD

**Issue by semmypurewal** [svaarala/duktape#2491]
Thanks for Duktape! I've had a lot of fun playing around with it as a hobby over the last couple years. This is the first issue I've stumbled upon.

In Duktape 2.7, it doesn't seem like I can set the `__proto__` property in an object literal. I haven't looked at the spec, but according to [this article](https://2ality.com/2015/09/proto-es6.html) it should be allowed. Here's an example that I would expect to evaluate to `true` but it evaluates to `false` in Duktape:

```
Object.getPrototypeOf({ __proto__ : null }) == null;
// => false
```

This does evaluate to `true` in V8.

This may seems like an edge case, but I found it because I'm writing [JSI bindings](https://github.com/facebook/hermes/tree/main/API/jsi/jsi) for Duktape 2.7. In one of [their tests](https://github.com/facebook/hermes/blob/main/API/jsi/jsi/test/testlib.cpp#L398-L413) they use this pattern.

**Comment by svaarala** [comment]
You're right, `__proto__` is at present not supported. Both `obj.__proto__` and object literal `__proto__` were specified as legacy features but in ES 13.0 `__proto__` in object literal seems to be a mainline feature (`obj.__proto__` is still legacy), so it needs to be added.
